bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,403 @@
++        exit -1
++    fi
++
++    req_files=("$CIV_WORK_DIR/OVMF.fd"
++                   "$CIV_WORK_DIR/scripts/rpmb_dev")
++    for file in ${req_files[@]}; do
++        if [ ! -f $file ]; then
++            echo "Error: $file file is missing"
++            exit -1
++        fi
++    done
++
++    cp $CIV_WORK_DIR/OVMF.fd $CIV_WORK_DIR/$VM_NAME/
++    cp $CIV_WORK_DIR/scripts/rpmb_dev $CIV_WORK_DIR/$VM_NAME/scripts/
++}
+ 
+ #-------------    main processes    -------------
+ 
+@@ -623,10 +669,12 @@ ubu_install_qemu_gvt
+ ubu_build_ovmf_gvt
+ ubu_enable_host_gvt
+ ubu_update_fw
+-
+ install_vm_manager
++
++create_vm_dir
+ create_aaf_dir
+-setup_civ_ini
++setup_civ_ini $VSOCK_ID $ADB_PORT $VM_NAME
++copy_files_for_vm
+ 
+ prepare_required_scripts
+ ubu_install_swtpm
+diff --git a/scripts/setup_multi_civ_vm.sh b/scripts/setup_multi_civ_vm.sh
+new file mode 100755
+index 0000000..370c681
+--- /dev/null
++++ b/scripts/setup_multi_civ_vm.sh
+@@ -0,0 +1,155 @@
++#!/bin/bash
++
++# Copyright (c) 2023 Intel Corporation.
++# All rights reserved.
++#
++# SPDX-License-Identifier: Apache-2.0
++
++set -eE
++
++#---------      Global variable     -------------------
++CIV_DIR=$(pwd)
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
++VM_NAME=Android-CIV
++VSOCK_ID=3
++NUMBER_OF_VM=1
++ADB_PORT=5555
++INITIAL_VM_NUM=1
++SETUP_HOST_DONE=false
++
++function copy_files_for_vm_flashing() {
++    echo "Copying file: $CIV_DIR/$3"
++    mkdir $CIV_DIR/$3/scripts
++
++    if [ -d "$CIV_DIR/scripts/aaf" ]; then
++        cp -r $CIV_DIR/scripts/aaf $CIV_DIR/$3/scripts/
++    else
++        echo "aaf folder missing. Please run setup_host.sh and try again"
++        exit -1
++    fi
++
++    req_files=("$CIV_DIR/OVMF.fd"
++                   "$CIV_DIR/scripts/rpmb_dev")
++    for file in ${req_files[@]}; do
++        if [ ! -f $file ]; then
++            echo "Error: $file file is missing"
++            exit -1
++        fi
++    done
++
++    cp $CIV_DIR/OVMF.fd $CIV_DIR/$3/
++    cp $CIV_DIR/scripts/rpmb_dev $CIV_DIR/$3/scripts/
++}
++
++function create_vm_dirs() {
++    echo "Dir: $CIV_DIR/$3"
++    mkdir $CIV_DIR/$3
++    $CIV_DIR/scripts/setup_civ_ini.sh -v $1 -p $2 -n $3
++}
++
++
++function create_vms() {
++    create_vm_dirs $1 $2 $3
++    copy_files_for_vm_flashing $1 $2 $3
++}
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -d $USER_HOME/.intel/.civ ]
++    then
++        SETUP_HOST_DONE=true
++    fi
++    new_dir=$CIV_DIR/$VM_NAME
++    if [ -n "$(ls -d ${new_dir}?)" ]; then
++	dir_arr=$(ls -d ${new_dir}?)
++	directories=( $dir_arr )
++	last_vm=${directories[-1]}
++	final_vm_name="${last_vm##*/}"
++	last_character=${final_vm_name: -1}
++	last_character=$((last_character+1))
++	ADB_PORT=$((5555+(last_character*2)))
++	VSOCK_ID=$((3+last_character))
++	INITIAL_VM_NUM=$last_character
++    fi
++
++    for (( i=$INITIAL_VM_NUM; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++	echo "Folder name: $CIV_DIR/${VM_NAME}${i}"
```

Comment:
Actual output:

> tester@tester-NUC:~$ sudo ./scripts/setup_multi_civ_vm.sh -c 2
> Creating setup for VMs
> ls: cannot access '/home/tester/Android-CIV?': No such file or directory
> Folder name: /home/tester/Android-CIV1
> Dir: /home/tester/Android-CIV1
> Copying file: /home/tester/Android-CIV1
> Folder name: /home/tester/Android-CIV2
> Dir: /home/tester/Android-CIV2
> Copying file: /home/tester/Android-CIV2
> Please reboot

I think there is no need to echo the "Folder name" since it is repeated in the next line